### DISCUSSION TITLE
Nested Pagination

### DISCUSSION THREAD

**Discussion by adliymeri** [11ty/eleventy#3027] [Q&A]
Hello everyone! 

I have a categories.njk template that creates a page for each category of my my markdown article files:

`---
title: Categories
layout: base.njk
pagination:
  data: collections.categories
  size: 1
  alias: currentCategory
  addAllPagesToCollections: true
permalink: "/categories/{{ currentCategory | slug }}/"
isCategoryPage: true
---`

Each article markdown file has a category in the front matter. I have also created a categories collection in my .eleventy.js file:

`eleventyConfig.addCollection("categories", function(collection) {
    let categorySet = new Set();
  
    collection.getAll().forEach(function(item) {
      if ('category' in item.data) {
        // Check if it's an array
        if (Array.isArray(item.data.category)) {
          item.data.category.forEach(function(cat) {
            categorySet.add(cat);
          });
        } else {
          // If it's just a single string, add it directly
          categorySet.add(item.data.category);
        }
      }
    });

    return [...categorySet];
  });
`
I want each category page created to also have pagination and show a set number of articles for each category page. The problem is I have already used pagination with my collections.categories to create pages for every category. Is there any way to do some sort of nested pagination, one for categories collection and one for the collections.article? My articles markdown files have the tags: article so the collections.article is already created.

**Comment by pdehaan** [comment]
See https://github.com/11ty/eleventy/issues/332 for some discussion/solutions.
